### PR TITLE
Added output_padding parameters in conv_transpose

### LINKED ISSUE #1423: [Feature] Add output_padding option to ConvTranspose2d
Passing the same parameters to `nn.ConvTranspose2d` does not revert `nn.Conv2d` in terms of dimension shape conservation, and an `output_padding` option can solve it.

Reference from [PyTorch's docs](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html):

> The padding argument effectively adds dilation * (kernel_size - 1) - padding amount of zero padding to both sizes of the input. This is set so that when a [Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d) and a [ConvTranspose2d](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html#torch.nn.ConvTranspose2d) are initialized with same parameters, they are inverses of each other in regard to the input and output shapes. However, when stride > 1, [Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html#torch.nn.Conv2d) maps multiple input shapes to the same output shape. output_padding is provided to resolve this ambiguity by effectively increasing the calculated output shape on one side. Note that output_padding is only used to find output shape, but does not actually add zero-padding to output.

### PR DESCRIPTION
**Pull request by ParamThakkar123**
## Proposed changes
Fixes #1423 

## Checklist

Put an `x` in the boxes that apply.

- [X] I have read the [CONTRIBUTING](https://github.com/ml-explore/mlx/blob/main/CONTRIBUTING.md) document
- [X] I have run `pre-commit run --all-files` to format my code / installed pre-commit prior to committing changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the necessary documentation (if needed)

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
@awni can you please review this PR ?

**Review by angeloskath — COMMENTED** [review]
This looks great to me! Do you mind adding it to the `nn.layers` and also add tests in `test_conv_transpose.py` as well, we already have a PyTorch dependent test there so it seems fitting.

**Comment by ParamThakkar123** [comment]
Sure @angeloskath . I will make these changes

**Comment by ParamThakkar123** [comment]
@angeloskath I added the output padding parameters to conv_transpose layers in `nn.layers` and also added the tests for them

**Review by angeloskath — APPROVED** [review]
I fixed the tests and docstrings, let's merge after the tests pass on CI.
